core/trino-main/src/main/java/io/trino/operator/output/SkewedPartitionRebalancer.java

```diff
@@ -347,9 +348,17 @@ private void rebalanceBasedOnTaskBucketSkewness(
                     break;
                 }
 
+                // Rebalance partition only once in a single cycle. Otherwise, rebalancing will happen quite
+                // aggressively in the early stage of write, while it is not required. Thus, it can have an impact on
+                // output file sizes and resource usage.
```

Comment:
nit: `output file sizes and resource usage (produced files can be small and memory usage might be higher)`.